stock_picking_force_assign/models/stock_picking.py

```diff
@@ -50,7 +56,7 @@ def _force_assign_find_moves(self):
             candidates = self.env['stock.move'].search([
                 ('id', 'not in', result.ids),
                 ('picking_id', 'not in', self.ids),
-                ('location_id', '=', location.id),
+                ('location_id', '=', move.location_id.id),
```

Comment:
why this?

Reply:
Because the location of the move and be different from the location of the picking, and in our case it often is. The methods down the line use the location of the move.

Reply:
oh, I thought it's actually wrong to have a picking with non-matching locations. But if that's the case, you need to change the test above to also look at the locations of the moves. But on the other hand I don't see the current logic relying on the location being the same everywhere anyways, so the whole test could go it seems

Reply:
Test removed.